### PR TITLE
docs: heap size and memory limit

### PR DESCRIPTION
**Pull request by PaulRMellor**
Signed-off-by: prmellor <pmellor@redhat.com>

**Documentation update**
Updates the [Common configuration properties](https://strimzi.io/docs/operators/in-development/configuring.html#con-common-configuration-properties-reference) section of the _Configuring_ guide. 

_**`-Xms` and `-Xmx` options**_ section rewritten to clarify heap size config in relation to memory limits. 

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: documentation/modules/con-common-configuration-properties.adoc [hunk 1]
```diff
@@ -115,7 +115,8 @@ For Kafka, the following aspects of a deployment can impact the resources you ne
 * The number of producers and consumers
 * The number of topics and partitions
 
-The values specified for resource requests are reserved and always available to a container.
+Resource requests set the resources a container needs to start.
```
Comment (scholzj) [inline documentation/modules/con-common-configuration-properties.adoc:118]: I would not use this. It is not completely true and it is misleading.

File: documentation/modules/con-common-configuration-properties.adoc [hunk 2]
```diff
@@ -514,7 +514,8 @@ Setting an appropriate memory request can prevent the following:
 If `-Xms` is set to `-Xmx`, the container will crash immediately; if not, the container will crash at a later time.
 
 In this example, the JVM uses 2 GiB (=2,147,483,648 bytes) for its heap.
-Its total memory usage is approximately 8GiB.
+Its total memory usage is approximately 8 GiB.
```
Comment (scholzj) [inline documentation/modules/con-common-configuration-properties.adoc:517]: I would leave this out. I don't think we have any data to back this number.

File: documentation/modules/con-common-configuration-properties.adoc [hunk 3]
```diff
@@ -499,13 +503,9 @@ The following table shows the default heap values.
 
 |===
 
-Heap configuration values should make the best use of a container's xref:con-common-configuration-resources-reference[memory request] without exceeding it.
-
-* If a memory request or limit is specified, a JVM's minimum and maximum memory is set to a value corresponding to the limit.
-* If a memory request or limit is not specified, a JVM's minimum memory is set to `128M`.
-The JVM's maximum memory is not defined to allow the memory to increase as needed. This is ideal for single node environments in test and development.
-
-Total JVM memory usage can be a lot more than the maximum heap size.
```
Comment (scholzj) [inline documentation/modules/con-common-configuration-properties.adoc:508]: I think thi is important nd should be kept.

File: documentation/modules/con-common-configuration-properties.adoc [hunk 4]
```diff
@@ -453,11 +454,14 @@ This is different from the units used for xref:con-common-configuration-resource
...
+Use the `-Xms` option to set an initial heap size and the `-Xmx` option to set a maximum heap size.
 
-If you don't specify heap configuration, but configure memory request or limit, the Cluster Operator imposes default heap sizes automatically.
-
-The Cluster Operator sets maximum and minimum heap values based on a percentage allocation of available memory in the node.
+Specify heap size to have more control over the memory allocated to your JVM.
+Heap sizes should make the best use of a container's xref:con-common-configuration-resources-reference[memory limit (and request)] without exceeding it.
+Heap size and any other memory requirements need to fit within a specified memory limit. 
+If you don't specify heap size in your configuration, but you configure a memory limit (and request), the Cluster Operator imposes default heap sizes automatically.
+The Cluster Operator sets default maximum and minimum heap values based on a percentage allocation of available memory in the node where the container is running.
```
Comment (scholzj) [inline documentation/modules/con-common-configuration-properties.adoc:464]: This is not true. If the resources are set, we set the heap as a percentage of the resource configuration. But we never set it based on the node memory.

File: documentation/modules/con-common-configuration-properties.adoc (hunk 1 above)
Comment (PaulRMellor) [reply documentation/modules/con-common-configuration-properties.adoc:118]: removed

File: documentation/modules/con-common-configuration-properties.adoc (hunk 2 above)
Comment (PaulRMellor) [reply documentation/modules/con-common-configuration-properties.adoc:517]: removed

File: documentation/modules/con-common-configuration-properties.adoc (hunk 4 above)
Comment (PaulRMellor) [reply documentation/modules/con-common-configuration-properties.adoc:464]: Now this makes much more sense! The node memory wasn't picked up in the last review, so I assumed that was how it was. Changed to:

> The Cluster Operator sets default maximum and minimum heap values based on a percentage of the memory resource configuration.

File: documentation/modules/con-common-configuration-properties.adoc (hunk 3 above)
Comment (PaulRMellor) [reply documentation/modules/con-common-configuration-properties.adoc:508]: We still have this. It's just no longer a bullet.

**Review by scholzj — APPROVED** [review]

**Pull request merged**